src/Sources.jl

```diff
@@ -107,6 +107,10 @@ function SetupSource(url::String, path::String, hash::String, target::String)
     end
 end
 
+struct PatchSource
+    name::String
+    patch::String
+end
```

Comment:
I'm a little worried someone will use this in their `build_tarballs.jl`; we probably want to give this a different name to make sure it's an internal-only thing.

Reply:
Note that it isn't an `AbstractSource` (nor it is exported), so someone using it as source in a builder won't go too far